### PR TITLE
Added news controllers

### PR DESCRIPTION
**Pull request by DikshantK2004**
Issue: #17 
### Changes
-  Added 4 routes and their corresponding controllers for news.
- Fixed the issue in the mongo Schema for news.
- added some comments in Schemas for easier understanding.
- improved responses in user controller (using response codes from the responseCodes.js)
- modified the adminMiddleware to satisfy the needs of news routes(because isAdmin route is called after validateToken which only gives access to id and not userName)
- remove redundant promise for validate Token

### Screenshots of the output

Screenshot from 2023-12-16 21-11-38
Screenshot from 2023-12-16 21-17-37
Screenshot from 2023-12-16 21-24-08
Screenshot from 2023-12-16 22-17-05
Screenshot from 2023-12-16 21-24-18

### CODE REVIEW

File: src/middlewares/auth.middleware.js [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 
 const validateToken = async (req, res, next) => {
   let token
-  let authHeader = req.headers.Authorization || req.headers.authorization
+  let authHeader = req.headers.Authorization || req.headers.authorization;
   if (authHeader && authHeader.startsWith("Bearer")) {
     token = authHeader.split(" ")[1]
     try {
-      const decoded = await verifyToken(token, process.env.JWT_KEY)
+      const decoded = jwt.verify(token, process.env.JWT_KEY)
```
Comment (Vihaan-19) [inline src/middlewares/auth.middleware.js:12]: @DikshantK2004 is there any specific reason to remove await here?

**Review by Vihaan-19 — CHANGES_REQUESTED** [review]
Good Work!

File: src/middlewares/auth.middleware.js (hunk 1 above)
Comment (DikshantK2004) [reply src/middlewares/auth.middleware.js:12]: No such reason, just that verify is a synchronous and someone had used a promisify to convert it to asynchronous but may be they had not imported something.  My VS code was not helping with it. So, I removed it.

Comment (Vihaan-19) [reply src/middlewares/auth.middleware.js:12]: @DikshantK2004 ok.

**Pull request merged**